internal/data/assets.go

```diff
@@ -46,6 +46,14 @@ func (a Asset) EqualsHorizonAsset(horizonAsset base.Asset) bool {
 	return strings.EqualFold(a.Code, horizonAsset.Code) && strings.EqualFold(a.Issuer, horizonAsset.Issuer)
 }
 
+func (a Asset) EqualsDistributionAccountMapID(mapID string) bool {
+	assetPair := strings.Split(mapID, ":")
+	if a.IsNative() {
+		return assetPair[0] == "XLM" && assetPair[1] == "native"
+	}
+	return a.Code == assetPair[0] && a.Issuer == assetPair[1]
+}
+
```

Comment:
We already have two other methods for comparison:
https://github.com/stellar/stellar-disbursement-platform-backend/blob/6bb32e9414de6df92413288b6a71107486ff47c1/internal/data/assets.go#L33-L47

MapID is to abstract of a concept to live here, don't you think?

Why do you use strings to represent assets, if we already have data.Asset for that?